### PR TITLE
Usage documentation

### PR DESCRIPTION
**Pull request by maxfischer2781**
This PR extends the documentation with concrete usage guidance. Major topics include:

* [x] extended client documentation with plaintext description
* [x] outline on how one could implement a ``class RemotePlayer``
* [x] short guide on connecting to and running the server

[There is a preview build on RTD.](https://battleship-mp.readthedocs.io/en/docs-usage/index.html)

### CODE REVIEW

File: docs/player.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+The constructor can then be used in a `with statement`_:
+
+.. code:: python3
+
+    with RemotePlayer.create(...) as player:
+        ...
+
+Note that you can create two players of type ``P1`` and ``P2``
+in one ``with`` statement using
+``with P1.create(...) as player_1, P2.create(...) as player_1:``.
```
Comment (giffels) [inline docs/player.rst]: Shouldn't it be `with P1.create(...) as player_1, P2.create(...) as player_2`?

**Review by giffels — CHANGES_REQUESTED** [review]
Except one typo it LGTM!

File: docs/player.rst (hunk 1 above)
Comment (maxfischer2781) [reply docs/player.rst]: Good catch, the numbering is consistent now.

**Review by giffels — DISMISSED** [review]
LGTM!

File: docs/client.rst [hunk 2]
```diff
@@ -8,6 +8,87 @@ Client Usage
...
+peers should :py:meth:`~.end_game` and announce the determined winner.
+This may be one of the players' identifiers or ``None`` in case of a draw.
+**The server validates whether there is a consensus of who won the game.**
+In addition, a player may :py:meth:`~.end_game` and ``forfeit`` the game at any time.
+
+Note that the client API – just like the server - generally imposes
+little restrictions on the game rules.
+If a restriction is not mentioned then the API does not impose it;
+for example, there is no restriction on the board size.
+Each client should check for itself whether all peer adhere to expected rules.
```
Comment (eileen-kuehn) [inline docs/client.rst]: Small typo here:

```suggestion
Each client should check for itself whether all peers adhere to expected rules.
```

File: docs/player.rst [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            self._enemy_ship_cache.append((size, pos, vertical))
+
+        def get_ship(self, size: int) -> "SHIP_PLACEMENT":
+            """Get the next placement for a ship of specific `size`"""
+            # send ship placement only when we need the response
+            if self._my_ship_buffer is None:
+                self._my_ship_buffer = list(
+                    session.place_ships(*self._enemy_ship_cache)
+                )
+                self._enemy_ship_cache = None
```
Comment (eileen-kuehn) [inline docs/player.rst]: I am confused a bit here. In `__init__` the `_enemy_ship_buffer` is created and in the following the `_enemy_ship_cache` is used. Do I oversee something?

**Review by eileen-kuehn — COMMENTED** [review]
I found another typo and was confused by some exemplary code. But maybe it's just the heat that makes the review more troublesome 😅

File: docs/player.rst (hunk 3 above)
Comment (maxfischer2781) [reply docs/player.rst]: Nice catch. I renamed both variables but without IDE support messed that up of course... 😅

**Pull request merged**